sdk/communication/communication-chat/test/public/chatThreadClient.spec.ts

```diff
@@ -2,44 +2,47 @@
 // Licensed under the MIT license.
 /* eslint-disable @typescript-eslint/no-invalid-this */
 
-import { isPlaybackMode, Recorder } from "@azure-tools/test-recorder";
+import { Recorder } from "@azure-tools/test-recorder";
 import { assert } from "chai";
 import { ChatClient, ChatThreadClient, ChatMessage } from "../../src";
 import { createTestUser, createRecorder, createChatClient } from "./utils/recordedClient";
 import { CommunicationIdentifier, getIdentifierKind } from "@azure/communication-common";
 import { Context } from "mocha";
+import { CommunicationUserToken } from "@azure/communication-identity";
 
 describe("ChatThreadClient", function () {
   let messageId: string;
   let recorder: Recorder;
   let chatClient: ChatClient;
   let chatThreadClient: ChatThreadClient;
   let threadId: string;
-
+  let communicationUserToken: CommunicationUserToken;
   let testUser: CommunicationIdentifier;
   let testUser2: CommunicationIdentifier;
   let testUser3: CommunicationIdentifier;
 
   beforeEach(async function (this: Context) {
-    recorder = createRecorder(this);
+    recorder = await createRecorder(this.currentTest);
+    if (!communicationUserToken) {
+      communicationUserToken = await createTestUser(recorder);
+      await recorder.setMatcher("HeaderlessMatcher");
```

Comment:
Headerless matcher matches the requests in playback without matching the headers in the recording.
Beyond the default skipped headers(to match), there was a new header specific to the communication service I believe, that was being dynamic. Hence using the headerless matcher.

There is going to be a custom-matcher soon from the test-proxy tool and the recorder, which will allow specifying the headers to be ignored while matching. Until then(even forever tbh), the Headerless matcher should be good and should do the job.

Reply:
Thanks for the explanation! Yes, our services return a dynamic ms-cv correlation vector.